src/senaite/core/browser/user/userdatapanel.py

```diff
@@ -183,22 +184,20 @@ def notify_linked_user(self):
             ))
 
         elif IContact.providedBy(contact):
-            fullname = contact.getFullname()
-            client_name = contact.aq_parent.getName()
-            self.add_status_message(_(
-                "User is linked to the client contact '${fullname}' "
-                "(${client_name})",
-                mapping={
-                    "fullname": api.safe_unicode(fullname),
-                    "client_name": api.safe_unicode(client_name)
-                }
-            ))
+            fullname = api.safe_unicode(contact.getFullname())
+            if contact.isGlobal():
+                msg = _("User is linked to the global contact '${fullname}'",
+                        mapping={"fullname": fullname})
```

Comment:
If the contact is global I would add this user to the client-specific group here. Likewise, I would remove the client-specific group when unlinked.